.github/workflows/cargo-publish.yaml

```diff
@@ -0,0 +1,30 @@
+name: Publish a release to crates.io
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
+
+jobs:
+  upload-crate:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v3
+
+    - name: cargo publish
+
+    
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - uses: actions-rs/cargo@v1
+      env:
+        CARGO_TOKEN: ${{ secrets.CARGO_TOKEN }}
+      with:
+        command: publish
+        args: --token $CARGO_TOKEN
```

Comment:
I think we need to publish each crate individually though, either by cd'ing into each crate and publishing it, or with `-p` on the root

Reply:
What's the manual release process currently like? I'll need to mirror that.

Reply:
right, sorry should have wrote that on the first comment. Basically what I do is in order release `refinery-core` first, then `refinery-macros` then `refinery` and lastly `refinery-cli` with:

```bash
$ cargo publish --all-features refinery-core
$ cargo publish --all-features refinery-macros
$ cargo publish --all-features refinery
$ cargo publish --all-features refinery_cli
```

Reply:
Check. That should be easy enough. I'll update the PR tomorrow.